include/albatross/src/models/ransac.hpp

```diff
@@ -147,7 +147,10 @@ RansacOutput ransac(const RansacFunctions<FitType> &ransac_functions,
   output.return_code = RANSAC_RETURN_CODE_FAILURE;
 
   if (min_consensus_size >= groups.size() ||
-      random_sample_size >= groups.size() || max_iterations <= 0) {
+      min_consensus_size < random_sample_size ||
+      random_sample_size >= groups.size() ||
+      random_sample_size <= 0 ||
```

Comment:
nit: Transitive makes the third equality redundant. :P
Looks good. No need to change.